### PR TITLE
DOCS-1842: Add handling for disk full

### PR DESCRIPTION
**Pull request by andf-viam**
Add notice of new full disk handling feature to Data management service > data capture reference page
- Also updated warning on Cloud sync page
- Also made tiny mention on Data ML tutorial

### CODE REVIEW

**Review by npentrel — APPROVED** [review]

**Review by npentrel — APPROVED** [review]

File: docs/data/capture/_index.md [hunk 1]
```diff
@@ -367,6 +369,35 @@ The following example captures data from the `ReadImage` method of a camera:
...
+
+If your machine captures a large amount of data, or frequently goes offline for long periods of time while capturing data, consider moving the Viam capture directory to a larger, dedicated storage device on your machine if available. You can change the capture directory using the `capture_dir` attribute.
+
+You can also control how local data is deleted if your machine's local storage becomes full, using the `delete_every_nth_when_disk_full` and `maximum_num_sync_threads` attributes:
+
+- The `delete_every_nth_when_disk_full` attribute controls how many files to delete when local storage meets the above fullness criteria.
+  The data management service will delete every Nth file that has been captured upon reaching this threshold.
+  The default value is `5`, meaning that every fifth captured file will be deleted.
+
+- The `maximum_num_sync_threads` attribute controls how many concurrent CPU threads to dedicate to the automatic deletion task when local storage meets the above fullness criteria.
```
[the remark this answers is not in the document]
Comment (andf-viam) [reply docs/data/capture/_index.md]: No worries; I think you're right then that it doesn't belong here. Stricken!

File: docs/data/capture/_index.md [hunk 2]
```diff
@@ -367,6 +369,35 @@ The following example captures data from the `ReadImage` method of a camera:
...
+  The data management service will use up to this configured number of threads.
+  The default value is `1000`, and is suitable for most cases.
+  Some Pi and Pi-based boards may encounter issues with the default thread count; if you encounter issues with automatic deletion, try raising this value as needed.
```
[the remark this answers is not in the document]
Comment (andf-viam) [reply docs/data/capture/_index.md]: Removed `maximum_num_sync_threads` instead.

**Comment by andf-viam** [comment]
@vijayvuyyuru Updated with your feedback! Let me know if you'd like so see any further changes -- thanks so much!

**Review by vijayvuyyuru — COMMENTED** [review]
All the file deletion information looks great! Thank you!
As far as the sync config information, could that go under a section in [configuration](https://docs-test.viam.dev/2905/data/cloud-sync/#configuration) here? That field is more of a "power user" feature. I think it would make sense if we add a section that says
Sync threading or threads?
Content could be something like: when sync is enabled, multiple threads will run concurrently to sync data to the viam cloud. By default, up to 1000 concurrent threads will run depending on how much data is synced locally. This value can be configured via the optional field `(i forgot what I named it)` in the data manager parameters.

The default value of 1000 is sufficient for most use cases.  If running on limited hardware, heavy load, and/or massive amounts of cached data, this value can be lowered.

I also think it would be nice to include an expanding section here for what the datamanger config can look like with this field set. 
Thanks for your work on this.

**Comment by andf-viam** [comment]
@vijayvuyyuru Sure thing, added `maximum_num_sync_threads` back in under its own dedicated section on cloud sync > configure. Thanks1

**Review by vijayvuyyuru — APPROVED** [review]
Looks great, thank you!!

**Pull request merged**